### PR TITLE
Enable account switcher

### PR DESCRIPTION
**Pull request by typfel**
This re-enables the team switcher for switching accounts. This is work-in-progress and many things still needs to be done.

depends on:
- [x] https://github.com/wireapp/wire-ios-sync-engine/pull/500

### CODE REVIEW

File: Wire-iOS/Sources/AppRootViewController.swift [hunk 1]
```diff
@@ -167,8 +185,8 @@ class AppRootViewController : UIViewController {
             MagicConfig.shared()
             self.installOverlayWindow()
         } else {
-            overlayWindow?.removeFromSuperview()
-            overlayWindow = nil
+//            overlayWindow?.removeFromSuperview()
+//            overlayWindow = nil
```
Comment (mikeger) [inline Wire-iOS/Sources/AppRootViewController.swift]: Not necessary? May be consider removing / adding a TODO

Comment (typfel) [inline Wire-iOS/Sources/AppRootViewController.swift]: Ah no this is proper TODO actually. Let me take another look at that.

Comment (typfel) [inline Wire-iOS/Sources/AppRootViewController.swift]: It's unnecessary to re-load magic when switching accounts but we probably want to remove the `NotificationStatusViewController` when logging out.

**Comment by mikeger** [comment]
FYI: I started the SE build

**Comment by mikeger** [comment]
wire-ios-sync-engine 105.1.1 is released

File: Wire-iOS/Sources/AppRootViewController.swift [hunk 2]
```diff
@@ -87,11 +101,41 @@ class AppRootViewController : UIViewController {
         sessionManager = SessionManager(appVersion: appVersion!, mediaManager: mediaManager!, analytics: analytics, delegate: appStateController, application: UIApplication.shared, launchOptions: launchOptions, blacklistDownloadInterval: Settings.shared().blacklistDownloadInterval)
     }
     
+    func enqueueTransition(to appState: AppState) {
+        
+        transitionQueue.async {
```
Comment (daehn) [inline Wire-iOS/Sources/AppRootViewController.swift]: Is it possible to put the `prepare` and `transition` calls into the same dispatch block?

File: Wire-iOS/Sources/AppStateController.swift [hunk 3]
```diff
@@ -76,6 +77,10 @@ class AppStateController : NSObject {
             return .blacklisted
         }
         
+        if (isSuspended) {
```
Comment (daehn) [inline Wire-iOS/Sources/AppStateController.swift]: `()` 😱

File: Wire-iOS/Sources/AppRootViewController.swift (hunk 2 above)
Comment (typfel) [reply Wire-iOS/Sources/AppRootViewController.swift]: Would we always enqueue the two block after another?

Comment (daehn) [reply Wire-iOS/Sources/AppRootViewController.swift]: Yeah, you're right, mixed something up.

**Pull request merged**